### PR TITLE
Add peer parameters to AnnouncePeers

### PR DESCRIPTION
**Pull request by mrd0ll4r**

### CODE REVIEW

File: server/store/memory/peer_store.go [hunk 1]
```diff
@@ -271,8 +271,14 @@ func (s *peerStore) AnnouncePeers(infoHash chihaya.InfoHash, seeder bool, numWan
 				}
 
 				if p.IP.To4() == nil {
+					if p.Equal(&peer6) {
+						continue
```
Comment (jzelinskie) [inline server/store/memory/peer_store.go]: numWant is never getting decremented when you do this continue.

Comment (mrd0ll4r) [inline server/store/memory/peer_store.go]: That's the idea - I don't add the announcing peer

Comment (jzelinskie) [inline server/store/memory/peer_store.go]: Oops -- was reading wayyy too fast. Should stop doing chihaya review during the workday 🙃

**Comment by jzelinskie** [comment]
LGTM

**Pull request merged**